### ISSUE TITLE
ConsKinkyPrefSolver doesn't need PrefShkPrbs

### ISSUE THREAD

**Issue by sbenthall** [econ-ark/HARK#641]
ConsKinkyPrefSolver, as a carry-over from a time without DiscreteDistribution, breaks a distribution into its probabilities and values before using them to construct a utility function:

https://github.com/econ-ark/HARK/blob/master/HARK/ConsumptionSaving/ConsPrefShockModel.py#L566-L567

https://github.com/econ-ark/HARK/blob/master/HARK/ConsumptionSaving/ConsPrefShockModel.py#L389-L390

This can be made more compact

**Comment by sbenthall** [comment]
I see. This coding style is quite widely used throughout HARK, not just in ConsPrefShockModel.

https://github.com/econ-ark/HARK/blob/master/HARK/ConsumptionSaving/ConsIndShockModel.py#L790-L793

I suppose that what happened was that because Distributions were once quite hard to understand (because they were just lists or arrays), it was helpful to copy their values to a more descriptively named one temporarily.

As we start to encapsulate distributions in objects, this should be less and less necessary. 

Note #623 -- we could use a more descriptive name for the values of a distribution than X.

Note also that Dolo uses an xarray, a data structure that allows for named columns and rows:

https://github.com/EconForge/dolo/issues/193

**Comment by sbenthall** [comment]
Depends on #730 now

**Comment by sbenthall** [comment]
This is a duplicate of #625 -- this all has to due with the somewhat convoluted expectation-taking code in the solutions, which is slated to be replaced with `calcExpectations`. See #896 for progress towards this.
